presentation/src/main/java/me/androidbox/presentation/agenda/screen/AgendaScreen.kt

```diff
@@ -160,44 +160,66 @@ fun AgendaScreen(
-                            .align(Alignment.BottomEnd),
-                        shouldOpenDropdown = agendaScreenState.shouldOpenEditAgendaDropdown,
-                        onCloseDropdown = {
-                            agendaScreenEvent(
-                                AgendaScreenEvent.OnChangeShowEditAgendaItemDropdownStatus(shouldOpen = false))
-                        },
-                        listOfMenuItemId = AgendaMenuActionType.values().map { it.titleId },
-                        onSelectedOption = { item ->
-                            agendaScreenState.agendaItemClicked?.let { agendaItem ->
-                                onSelectedEditAgendaItemClicked(agendaItem, AgendaMenuActionType.values()[item])
-                                agendaScreenEvent(AgendaScreenEvent.OnChangeShowEditAgendaItemDropdownStatus(shouldOpen = false))
+                    Box(modifier = Modifier.fillMaxWidth()) {
+                        AgendaCard(
+                            modifier = Modifier
+                                .fillMaxWidth()
+                                .height(120.dp)
+                                .clip(shape = RoundedCornerShape(22.dp)),
+                            title = agendaItem.title,
+                            subtitle = agendaItem.description,
+                            dateTimeInfo = agendaItem.toDisplayDateTime(),
+                            agendaCardType = when (agendaItem.agendaType) {
+                                AgendaType.EVENT -> AgendaCardType.EVENT
+                                AgendaType.TASK -> AgendaCardType.TASK
+                                AgendaType.REMINDER -> AgendaCardType.REMINDER
+                            },
+                            isAgendaCompleted = false,
+                            onMenuOptionClicked = {
+                                println("Event ${agendaItem.id} has been clicked")
+                                agendaScreenEvent(AgendaScreenEvent.OnAgendaItemClicked(agendaItem))
+                                agendaScreenEvent(
+                                    AgendaScreenEvent.OnChangeShowEditAgendaItemDropdownStatus(
+                                        shouldOpen = true,
+                                        agendaItem = agendaItem
+                                    )
+                                )
+                            },
+                            dropDownMenu = {
+                                /** Open, Edit, Delete Agenda Items */
+                                AgendaDropDownMenu(
+                                    modifier = Modifier
+                                        .background(color = MaterialTheme.colorScheme.dropDownMenuBackgroundColor)
+                                        .align(Alignment.BottomEnd),
+                                    shouldOpenDropdown = agendaItem == agendaScreenState.agendaItemWithOperationBeingSelected && agendaScreenState.shouldOpenEditAgendaDropdown,
```

Comment:
@philipplackner I have implemented based on your solution. Is this correct?

Reply:
The 2nd state isn't necessary since you only show the drop down if `agendaItemWithOperationBeingSelected` is not null. If it's null, no drop down is shown.